cloudinstall/controllers/installbase.py

```diff
@@ -110,9 +112,8 @@ def update(self, *args, **kwargs):
             self.ui.render_machine_wait_view(self.config)
             self.loop.redraw_screen()
 
-        alarm = self.loop.set_alarm_in(1, self.update)
-        self.observe('stop alarm', partial(self.loop.remove_alarm,
-                                           alarm))
+        AlarmMonitor.add_alarm(self.loop.set_alarm_in(1, self.update))
+        self.observe('stop alarm', AlarmMonitor.abort)
```

Comment:
I wonder if there's a cleaner way to do this - maybe InstallController doesn't have to be an Observer at all, and maybe AlarmMonitor can just watch for the events itself. So here we'd just call AlarmMonitor.add_alarm() and then that'd handle setting up a watch for the stop alarm event, which would then do the abort itself.

What do you think?